tests/components/lektrico/test_config_flow.py

```diff
@@ -0,0 +1,274 @@
+"""Tests for the Lektrico Charging Station config flow."""
+
+import dataclasses
+from ipaddress import ip_address
+from unittest.mock import patch
+
+from lektricowifi import DeviceConnectionError
+
+from homeassistant import config_entries
+from homeassistant.components.lektrico.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import (
+    ATTR_HW_VERSION,
+    ATTR_SERIAL_NUMBER,
+    CONF_FRIENDLY_NAME,
+    CONF_HOST,
+    CONF_TYPE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .conftest import (
+    MOCKED_DEVICE_BAD_ID_ZEROCONF_DATA,
+    MOCKED_DEVICE_BAD_NO_ID_ZEROCONF_DATA,
+    MOCKED_DEVICE_BOARD_REV,
+    MOCKED_DEVICE_FRIENDLY_NAME,
+    MOCKED_DEVICE_IP_ADDRESS,
+    MOCKED_DEVICE_SERIAL_NUMBER,
+    MOCKED_DEVICE_SERIAL_NUMBER_FOR_3EM,
+    MOCKED_DEVICE_SERIAL_NUMBER_FOR_EM,
+    MOCKED_DEVICE_TYPE,
+    MOCKED_DEVICE_TYPE_FOR_3EM,
+    MOCKED_DEVICE_TYPE_FOR_EM,
+    MOCKED_DEVICE_ZEROCONF_DATA,
+    MOCKED_DEVICE_ZEROCONF_DATA_FOR_3EM,
+    MOCKED_DEVICE_ZEROCONF_DATA_FOR_EM,
+    patch_device_config,
+    patch_device_config_for_3em,
+    patch_device_config_for_em,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_user_setup(hass: HomeAssistant) -> None:
+    """Test manually setting up."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
+    assert "flow_id" in result
+
+    with (
+        patch_device_config(),
```

Comment:
How does this work considering `patch_device_config` is not a context manager?

Reply:
Yes, you're right, it will work 👍 
However, we don't usually make test helpers which return patches, we instead use pytest fixtures. I want a second opinion on if this is OK or not.